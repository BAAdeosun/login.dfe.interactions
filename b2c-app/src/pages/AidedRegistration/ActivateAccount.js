import React from 'react';
import components from '../../components';

class ActivateAccount extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            password: null,
            dobDay: null,
            dobMonth: null,
            dobYear: null,
            termsAndConditionsAccepted: false
        };
    }

    onPasswordChange = (value) => {
        this.setState({ password: value }, () => {
            console.log('password in page set to ' + this.state.password);
        });
    }

    render() {

        return (
            <div id="activateAccount">

                <div className="govuk-width-container">
                    <components.Breadcrumbs />

                    <components.PageLevelErrorContainer summaryTextContent={<components.PasswordHelp />} />

                    <main className="govuk-main-wrapper">
                        <div className="govuk-grid-row">
                            <div className="govuk-grid-column-two-thirds">
                                <components.PageTitle size='xl' title="Activate your account" />

                                <form id="activateAccountForm" noValidate>
                                    <components.CreateNewPassword onChange={this.onPasswordChange} />
                                    <p className="govuk-body">As an extra security check, enter your date of birth.</p>
                                    <components.DateOfBirthInput />
                                    <components.TermsAndConditions />
                                    <button className="govuk-button" id="preSubmit" type="submit">Activate account</button>
                                </form>

                                {/* add B2C component, it will be hidden but needs to be loaded */}
                                {/* TODO move to index.html and remove from all pages */}
                                <div id="api" />
                            </div>
                        </div>
                    </main>

                </div>

                <script src="__--b2cPath--__/b2c/assets/js-static/pages/aidedRegistration/activateAccount.js"></script>
                <script src="__--b2cPath--__/b2c/assets/js-static/validation/new-password.js"></script>
                <script src="__--b2cPath--__/b2c/assets/js-static/validation/date-of-birth.js"></script>
                <script src="__--b2cPath--__/b2c/assets/js-static/validation/terms-and-conditions.js"></script>

            </div>
        )
    }
}

export default ActivateAccount;