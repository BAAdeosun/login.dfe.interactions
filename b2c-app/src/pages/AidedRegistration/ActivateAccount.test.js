import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import ActivateAccount from './ActivateAccount';

it('renders without crashing', () => {
    shallow(<ActivateAccount />);
});

it('renders correctly', () => {
    const tree = renderer
        .create(<ActivateAccount />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});