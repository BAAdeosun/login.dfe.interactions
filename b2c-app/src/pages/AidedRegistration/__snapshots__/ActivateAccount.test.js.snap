// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="activateAccount"
>
  <div>
    <div
      className="govuk-width-container"
    >
      <div
        className="govuk-breadcrumbs"
      >
        <ol
          className="govuk-breadcrumbs__list"
        >
          <li
            className="govuk-breadcrumbs__list-item"
          >
            <a
              className="govuk-breadcrumbs__link"
              href="https://nationalcareers.service.gov.uk/"
            >
              Home
            </a>
          </li>
        </ol>
      </div>
      <main
        className="govuk-main-wrapper"
      >
        <div
          className="govuk-grid-row"
        >
          <div
            className="govuk-grid-column-two-thirds"
          >
            <div
              className="pageLevelErrorContainer hide"
            >
              <div
                aria-labelledby="error-summary-title"
                className="govuk-error-summary"
                data-module="govuk-error-summary"
                role="alert"
                tabIndex="-1"
              >
                <h2
                  className="govuk-error-summary__title"
                  id="error-summary-title"
                >
                  There is a problem
                </h2>
                <div
                  className="govuk-error-summary__body"
                >
                  <ul
                    className="govuk-list govuk-error-summary__list"
                    id="errorSummaryItems"
                  />
                </div>
              </div>
            </div>
            <h1
              className="govuk-heading-xl govuk-!-margin-top-6 govuk-!-margin-bottom-9"
            >
              Activate your account
            </h1>
            <form
              id="customForm"
              noValidate={true}
              onSubmit={[Function]}
            >
              <div>
                <div>
                  <div
                    className="govuk-form-group "
                  >
                    <label
                      className="govuk-label"
                      htmlFor="newPasswordCustom"
                    >
                      Create new password
                    </label>
                    <input
                      className="govuk-input govuk-!-width-one-half"
                      id="newPasswordCustom"
                      name="newPassword"
                      noValidate={true}
                      onChange={[Function]}
                      type="password"
                    />
                  </div>
                  <div
                    className="govuk-form-group"
                  >
                    <details
                      className="govuk-details"
                    >
                      <summary
                        className="govuk-details__summary"
                      >
                        <span
                          className="govuk-details__summary-text"
                        >
                          Help choosing a valid password
                        </span>
                      </summary>
                      <div
                        className="govuk-details__text"
                      >
                        <div>
                          <p
                            className="govuk-body"
                          >
                            Your password must:
                          </p>
                          <ul
                            className="govuk-list govuk-list--bullet"
                          >
                            <li>
                              be between 8 and 16 characters
                            </li>
                            <li>
                              include at least one uppercase letter and one lowercase letter
                            </li>
                            <li>
                              include at least one number or one symbol
                            </li>
                          </ul>
                        </div>
                      </div>
                    </details>
                  </div>
                  <div
                    className="govuk-form-group "
                  >
                    <label
                      className="govuk-label"
                      htmlFor="reenteredPasswordCustom"
                    >
                      Re-type password
                    </label>
                    <input
                      className="govuk-input govuk-!-width-one-half"
                      id="reenteredPasswordCustom"
                      name="reenteredPassword"
                      noValidate={true}
                      onChange={[Function]}
                      type="password"
                    />
                  </div>
                </div>
                <p
                  className="govuk-body"
                >
                  As an extra security check, please enter your date of birth.
                </p>
                <div
                  className="govuk-form-group "
                >
                  <fieldset
                    aria-describedby="date-of-birth-hint"
                    className="govuk-fieldset"
                    id="dobFieldset"
                    role="group"
                  >
                    <legend
                      className="govuk-fieldset__legend govuk-fieldset__legend--l"
                    >
                      <label
                        className="govuk-label"
                      >
                        Date of birth
                      </label>
                    </legend>
                    <span
                      className="govuk-hint"
                      id="date-of-birth-hint"
                    >
                      For example, 31 3 1980
                    </span>
                    <div
                      className="govuk-date-input"
                      id="date-of-birth"
                    >
                      <div
                        className="govuk-date-input__item"
                      >
                        <div
                          className="govuk-form-group"
                        >
                          <label
                            className="govuk-label govuk-date-input__label"
                            htmlFor="dobDay"
                          >
                            Day
                          </label>
                          <input
                            className="govuk-input govuk-date-input__input govuk-input--width-2"
                            id="dobDay"
                            inputMode="numeric"
                            name="dobDay"
                            noValidate={true}
                            onChange={[Function]}
                            pattern="[0-9]*"
                            type="number"
                          />
                        </div>
                      </div>
                      <div
                        className="govuk-date-input__item"
                      >
                        <div
                          className="govuk-form-group"
                        >
                          <label
                            className="govuk-label govuk-date-input__label"
                            htmlFor="dobMonth"
                          >
                            Month
                          </label>
                          <input
                            className="govuk-input govuk-date-input__input govuk-input--width-2"
                            id="dobMonth"
                            inputMode="numeric"
                            name="dobMonth"
                            noValidate={true}
                            onChange={[Function]}
                            pattern="[0-9]*"
                            type="number"
                          />
                        </div>
                      </div>
                      <div
                        className="govuk-date-input__item"
                      >
                        <div
                          className="govuk-form-group"
                        >
                          <label
                            className="govuk-label govuk-date-input__label"
                            htmlFor="dobYear"
                          >
                            Year
                          </label>
                          <input
                            className="govuk-input govuk-date-input__input govuk-input--width-4"
                            id="dobYear"
                            inputMode="numeric"
                            name="dobYear"
                            noValidate={true}
                            onChange={[Function]}
                            pattern="[0-9]*"
                            type="number"
                          />
                        </div>
                      </div>
                    </div>
                  </fieldset>
                </div>
                <div>
                  <h1
                    className="govuk-heading-m"
                  >
                    Terms and conditions
                  </h1>
                  <div
                    className="govuk-form-group "
                  >
                    <label
                      className="block-label"
                      htmlFor="tsAndCsCustom"
                    >
                      <input
                        aria-invalid="true"
                        checked={false}
                        id="tsAndCsCustom"
                        name="tsAndCsAccepted"
                        noValidate={true}
                        onChange={[Function]}
                        type="checkbox"
                        value={true}
                      />
                      I accept the 
                      <a
                        href="https://nationalcareers.service.gov.uk/help/terms-and-conditions"
                        id="tsAndCsLink"
                        rel="noopener noreferrer"
                        target="_blank"
                      >
                        terms and conditions
                      </a>
                       and I am 13 or over
                    </label>
                  </div>
                </div>
              </div>
              <button
                className="govuk-button"
                id="preSubmit"
                type="submit"
              >
                Activate account
              </button>
            </form>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;
